samples/ar-view-node/src/main/java/io/github/sceneview/sample/arviewnode/MainActivityViewMvcImpl.kt

```diff
@@ -0,0 +1,133 @@
+package io.github.sceneview.sample.arviewnode
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import io.github.sceneview.ar.ARSceneView
+import io.github.sceneview.sample.arviewnode.databinding.ActivityMainBinding
+import java.util.Collections
+
+class MainActivityViewMvcImpl(
+    private val context: Context,
+    inflater: LayoutInflater,
+    parent: ViewGroup?,
+) : View.OnClickListener, MainActivityViewMvc {
+    private val binding: ActivityMainBinding
+
+    private val _listeners: MutableSet<MainActivityViewMvc.Listener> = HashSet()
+
+    private val listeners: Set<MainActivityViewMvc.Listener>
+        get() = Collections.unmodifiableSet(_listeners)
+
+    ///////
+    // region Init
+    init {
+        binding = ActivityMainBinding
+            .inflate(
+                inflater,
+                parent,
+                false
+            )
+    }
+
+    init {
+        binding.btnAdd.setOnClickListener(this)
+        binding.btnViewNode.setOnClickListener(this)
+        binding.btnImageNode.setOnClickListener(this)
+    }
+    // endregion Init
+    ///////
+
+
+    ///////////////////////
+    // region View.OnClickListener
+    override fun onClick(v: View) {
+
+        when (v.id) {
+            binding.btnAdd.id -> {
+                listeners.forEach {
+                    it.handleOnAddClicked()
+                }
+            }
+
+            binding.btnViewNode.id -> {
+                handleOnSetViewNodeClicked()
+            }
+
+            binding.btnImageNode.id -> {
+                handleOnSetImageNodeClicked()
+            }
+        }
+    }
+
+    private fun handleOnSetViewNodeClicked() {
+        listeners.forEach {
+            it.updateActiveNodeType(MainActivity.NodeType.VIEW_NODE)
+        }
+
+        applyActiveColor(binding.btnViewNode)
+        applyDisabledColor(binding.btnImageNode)
+    }
+
+    private fun handleOnSetImageNodeClicked() {
```

Comment:
I'd rather call this method `onSetImageNodeClick` or `handleSetImageNodeClick`.

Reply:
I can go with that